### PR TITLE
New Toggle Component.

### PR DESCRIPTION
**Pull request by JeffBue**
Please review this toggle switch

### CODE REVIEW

File: src/elements/Toggle/Toggle.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+const Toggle = ({
+	checked,
+	value,
+	loading,
+	...other
+}) => {
+	const combinedClasses = classNames(
+		'ui-toggle',
+		other.classes,
+		checked && 'on',
```
Comment (jfurfaro) [inline src/elements/Toggle/Toggle.js]: When using `classNames`, you can put an object in any argument position where the keys are the class names, and the values evaluate to truthy/falsy to turn those class names on. It's much more readable when you have long lists of classes to toggle:

```
classNames('ui-component', {
  'some-class': someCheck,
  on: checked,
  loading
})
```

File: src/elements/Toggle/toggle.less [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	    width: 26px;
+	    height: 26px;
+	    border-radius: 100%;
+	    box-shadow: 0 2px 1.96px 0.04px rgba(0,0,0,.2);
+	    transition: left .2s ease-in-out, background .1s ease-in-out;
+	}
+
+	&.loading {
+		opacity: 0.5;
+		cursor: not-allowed;
```
Comment (zepfan) [inline src/elements/Toggle/toggle.less]: Delete this line (the next `cursor` property overwrites it)

File: src/elements/Toggle/Toggle.js (hunk 1 above)
Comment (JeffBue) [reply src/elements/Toggle/Toggle.js]: So like this:
```
	const combinedClasses = classNames(
		'ui-toggle',
		other.classes,
		{
			'on': checked,
			loading
		}
	);
```

File: src/elements/Toggle/toggle.less (hunk 2 above)
Comment (JeffBue) [reply src/elements/Toggle/toggle.less]: yeah thats weird how that remained in there, good catch thanks!

File: src/elements/Toggle/Toggle.js (hunk 1 above)
Comment (jfurfaro) [reply src/elements/Toggle/Toggle.js]: yessir!

File: src/elements/Toggle/Toggle.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+			on: checked,
+			loading
+		}
+	);
+
+	return (
+		<Common
+			{...other}
+			classes={combinedClasses}
+			tag="div"
```
Comment (zepfan) [inline src/elements/Toggle/Toggle.js]: This should actually be an `input` with `type` of `checkbox` (this change will probably require a couple style changes)

File: src/elements/Toggle/Toggle.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+			data-checked={checked}
+			data-value={value}
```
Comment (zepfan) [inline src/elements/Toggle/Toggle.js]: Once you change this to an `input`, you won't need these data attributes anymore

File: src/elements/Toggle/Toggle.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+		/>
+	);
+};
+
+Toggle.propTypes = {
```
Comment (zepfan) [inline src/elements/Toggle/Toggle.js]: You're missing the `value` prop validation

File: src/elements/Toggle/Toggle.js (hunk 3 above)
Comment (JeffBue) [reply src/elements/Toggle/Toggle.js]: Changed type to checkbox, no warnings or errors.

File: src/elements/Toggle/Toggle.js (hunk 4 above)
Comment (JeffBue) [reply src/elements/Toggle/Toggle.js]: removed data- attributes.

File: src/elements/Toggle/Toggle.js (hunk 5 above)
Comment (JeffBue) [reply src/elements/Toggle/Toggle.js]: added value prop validation.

File: src/elements/Toggle/Toggle.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import Common from '../Common/Common';
+
+const Toggle = ({
+	checked,
+	value,
+	loading,
```
[the remark this answers is not in the document]
Comment (JeffBue) [reply src/elements/Toggle/Toggle.js]: Changed loading to disabled.

**Comment by jfurfaro** [comment]
Closing this for a new updated PR from a fresh branch as this is stale

**Pull request closed without merging**